sct-commons/src/main/java/org/lfenergy/compas/sct/commons/scl/ConnectedApService.java

```diff
@@ -0,0 +1,68 @@
+// SPDX-FileCopyrightText: 2023 RTE FRANCE
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package org.lfenergy.compas.sct.commons.scl;
+
+import org.lfenergy.compas.scl2007b4.model.*;
+
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+
+import static org.lfenergy.compas.sct.commons.util.SclConstructorHelper.newAddress;
+
+public class ConnectedApService {
+
+    public Optional<TConnectedAP> findConnectedAp(TCommunication tCommunication, String iedName, String apName) {
+        if (!tCommunication.isSetSubNetwork()) {
+            return Optional.empty();
+        }
+        return tCommunication.getSubNetwork().stream()
+                .filter(TSubNetwork::isSetConnectedAP)
+                .flatMap(tSubNetwork -> tSubNetwork.getConnectedAP().stream())
+                .filter(tConnectedAP -> iedName.equals(tConnectedAP.getIedName()) && apName.equals(tConnectedAP.getApName()))
+                .findFirst();
+    }
+
+    /**
+     * Create A GSE Section or update an existing GSE Section (the network configuration of a GSEControl block).
+     */
+    public void updateGseOrCreateIfNotExists(TConnectedAP tConnectedAP, String ldInst, String cbName, List<TP> listOfP, TDurationInMilliSec minTime, TDurationInMilliSec maxTime) {
+        Optional<TGSE> optGse = tConnectedAP.isSetGSE() ?
+                tConnectedAP.getGSE().stream().filter(gse1 -> Objects.equals(ldInst, gse1.getLdInst()) && Objects.equals(cbName, gse1.getCbName())).findFirst()
+                : Optional.empty();
+        TGSE gse = optGse
+                .orElseGet(() -> {
+                            TGSE newGse = new TGSE();
+                            newGse.setLdInst(ldInst);
+                            newGse.setCbName(cbName);
+                            tConnectedAP.getGSE().add(newGse);
+                            return newGse;
+                        }
+                );
+        gse.setAddress(newAddress(listOfP));
+        gse.setMinTime(minTime);
+        gse.setMaxTime(maxTime);
+    }
+
+    /**
+     * Create A SMV Section or update an existing SMV Section (the network configuration of a SampledValueControl block)..
+     */
+    public void updateSmvOrCreateIfNotExists(TConnectedAP tConnectedAP, String ldInst, String cbName, List<TP> listOfP) {
```

Comment:
Such operation can be done by Editor suffixed service ?

Reply:
Same answer as `updateGseOrCreateIfNotExists`